File: leetcode-top-interview-150/198.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        lth = len(nums)

        if lth <= 2:
            return max(nums)

        dp = [0] * lth

        dp[0] = nums[0]
        dp[1] = max(nums[0], nums[1])

        for i in range(2, lth):
            dp[i] = max(dp[i - 2] + nums[i], dp[i - 1])

        return dp[-1]

File: leetcode-75/198.kt
class Solution {
    fun rob(nums: IntArray): Int {
    
        if (nums.size <= 2) {
            return nums.maxOrNull() ?: 0
        }

        var dp = Array<Int>(nums.size){0}

        dp[0] = nums[0]
        dp[1] = Math.max(nums[0], nums[1])

        for(i in 2..nums.size-1) {
            dp[i] = maxOf(dp[i-2]+nums[i], dp[i-1])
        }

        return dp.last()
    }
}